ECK quickstart access endpoint, curl and TLS verifications (#6566)
elastic/docs-content: deploy-manage/deploy/cloud-on-k8s/elasticsearch-deployment-quickstart.md

quickstart    green     1         {{version.stack}}     Ready         1m
```

## Access your {{es}} cluster [k8s_request_es_access]

> ECK automatically creates a `ClusterIP` Service for [HTTP access to your cluster](/deploy-manage/deploy/cloud-on-k8s/accessing-services.md#k8s-kubernetes-service). You can verify it with `kubectl get`:

Reviewer: ```suggestion
ECK automatically creates a `ClusterIP` service for [HTTP access to your cluster](/deploy-manage/deploy/cloud-on-k8s/accessing-services.md#k8s-kubernetes-service). You can verify it with `kubectl get`:
```

    PASSWORD=$(kubectl get secret quickstart-es-elastic-user -o go-template='{{.data.elastic | base64decode}}')
    ```

1. Retrieve the CA certificate.

    By default, ECK enables HTTPS for {{es}}, generates a private CA for each cluster, and issues certificates signed for the associated DNS service names, such as `quickstart-es-http.<namespace>.svc`.

    The CA certificate is available in the `<name>-es-http-certs-public` secret. For this `quickstart` cluster, run the following command to save the CA certificate to a local file named `quickstart-es-ca.crt`:

    ```sh
    kubectl get secret quickstart-es-http-certs-public -o go-template='{{index .data "ca.crt" | base64decode }}' > quickstart-es-ca.crt
    ```

    Refer to [Manage HTTP certificates on ECK](/deploy-manage/security/k8s-https-settings.md) for information about customizing HTTP TLS configuration.

> 1. Request the [{{es}} root API]({{es-apis}}group/endpoint-info). You can do so from inside the {{k8s}} cluster or from your local workstation.

Reviewer: ```suggestion
1. Issue a request to the [{{es}} info API]({{es-apis}}group/endpoint-info). You can do so from inside the {{k8s}} cluster or from your local workstation.
```

Reviewer: (in the linked API doc, this is called the info API, though i know it's also the root endpoint in a general sense... feel free to revise!)

    For testing only, you can use [`--insecure`](https://curl.se/docs/manpage.html#-k) (or `-k`) to skip certificate verification. This flag turns off TLS trust checks and should not be used in production.
    :::

    * **From inside the Kubernetes cluster**

        Use the service name to access the {{es}} endpoint from any {{k8s}} Pod:

        ```sh
        curl --cacert <PATH_TO_CA> -u "elastic:$PASSWORD" "https://quickstart-es-http.<namespace>.svc:9200" <1>
        ```
        1. Replace `<namespace>` with the namespace where your {{es}} cluster is deployed.

        For example, if you run the command from an {{es}} Pod and the cluster is deployed in the `default` namespace, you can run:

        ```sh
        curl --cacert /usr/share/elasticsearch/config/http-certs/ca.crt \
          -u "elastic:$PASSWORD" "https://quickstart-es-http.default.svc:9200"
        ```

    * **From your local workstation**

        1. Start a local port-forward in a separate terminal to route `localhost:9200` to the `quickstart-es-http` {{k8s}} Service:

            ```sh
            kubectl port-forward service/quickstart-es-http 9200
            ```

            :::{note}

>             Port-forwarding is mainly intended for local testing. In production environments, if the cluster must be accessible from outside the {{k8s}} cluster, consider using a `LoadBalancer` Service or another exposure mechanism. Refer to [Allow public access](/deploy-manage/deploy/cloud-on-k8s/accessing-services.md#k8s-allow-public-access) for more information.

Reviewer: ```suggestion
            Port-forwarding is mainly intended for local testing. In production environments, if the cluster must be accessible from outside the {{k8s}} cluster, consider using a `LoadBalancer` service or another exposure mechanism. Refer to [Allow public access](/deploy-manage/deploy/cloud-on-k8s/accessing-services.md#k8s-allow-public-access) for more information.
```
